libs/remix-ui/editor/src/lib/remix-ui-editor.tsx

```diff
@@ -541,6 +543,33 @@ export const EditorUI = (props: EditorUIProps) => {
       }
     })
 
+    editor.onDidPaste((e) => {
+       if (!pasteCodeRef.current && e && e.range && e.range.startLineNumber >= 0 && e.range.endLineNumber >= 0 && e.range.endLineNumber - e.range.startLineNumber > 10) {
+        const modalContent: AlertModal = {
+          id: 'newCodePasted',
+          title: 'Pasted code Alert',
+          message: (
+            <div> <i className="fas fa-exclamation-triangle text-danger mr-1"></i>
+              You have just pasted a code snippet or contract in the editor.
+              <div>
+                Make sure you fully understand this new code before deploying or interacting with it.
+                <div>
+                Your wallet might be exposed and <span className='text-warning'> at risk </span> if you run untrusted code. In a worst-case scenario, you could <span className='text-warning'>loose all your money</span>.
```

Comment:
Here's an edit - but leave the text-warning spans.
Running untrusted code can put your wallet at risk. In a worst-case scenario, you could loose all your money.